### ISSUE TITLE
Nested class within a template class could not be auto-completated

### ISSUE THREAD

**Issue by median-dxz** [microsoft/vscode-cpptools#8937]
Bug type: Language Service

**Describe the bug**
- OS and Version:  Windows_NT x64 10.0.22000
- VS Code Version: 1.65.0-insider (user setup)
- C/C++ Extension Version: v1.9.1
- Other extensions you installed (and if the issue persists after disabling them): persists upon disabling all extensions.

description: When a template class with a nested class is derived, the nested class could not be auto-completated by Intellisense in this derived class.

**Steps to reproduce**

Just use following code.

**Expected behavior**

The auto-completion pop-up should display something.

**Code sample and logs**
```cpp
#include <iostream>

template <typename T> struct A {
    struct B {
        T member;
    };
};

template <typename T> struct C : public A<T> {
    void D(T val) {
        typename A<T>::B a;
        //a.
        a.member = val;
        std::cout << a.member << std::endl;
    }
};

int main() {
    C<int> a;
    a.D(10);
    return 0;
}
```

**Screenshots**
![image](https://user-images.githubusercontent.com/33348142/155835109-8dbda5d5-8977-458e-aea0-1a1932065eee.png)

**Comment by sean-mcmanus** [comment]
This seems like a duplicate of https://github.com/microsoft/vscode-cpptools/issues/4920. Waiting for confirmation on that...

**Comment by sean-mcmanus** [comment]
This is "by design" because a template specialization could exist that doesn't have a "member" defined, e.g.
```
template <>
struct A<int> {
    struct B{};
};
```

Maybe this could be improved after we allow templated types to be specified, see https://github.com/microsoft/vscode-cpptools/issues/1946 .

**Comment by sean-mcmanus** [comment]
To be more clear, it's theoretically possible we could implement a change to make this work, but our current plan (design) is to fix https://github.com/microsoft/vscode-cpptools/issues/1946 instead.

**Comment by median-dxz** [comment]
> To be more clear, it's theoretically possible we could implement a change to make this work, but our current plan (design) is to fix #1946 instead.

Ok, thanks for your reply.
